routers/api/v1/repo/issue.go

```diff
@@ -81,8 +81,10 @@ func SearchIssues(ctx *context.APIContext) {
 		AllPublic:   true,
 		TopicOnly:   false,
 		Collaborate: util.OptionalBoolNone,
-		OrderBy:     models.SearchOrderByRecentUpdated,
-		Actor:       ctx.User,
+		// This needs to be a column that is not nil in fixtures or
+		// MySQL will return different results when sorting by null in some cases
+		OrderBy: models.SearchOrderByAlphabetically,
```

Comment:
I'm afraid that this would be considered a breaking change. My suggestion, rather than changing the API endpoint behavior, would be to update the fixtures and add the missing values in the columns (which should have to start with). Other than that, we may be able to add the alphabetic _order by_ as a secondary key in columns that can be null. e.g.:

```
// Strings for sorting result
const (
    SearchOrderByAlphabetically        SearchOrderBy = "name ASC"
    SearchOrderByAlphabeticallyReverse SearchOrderBy = "name DESC"
    SearchOrderByLeastUpdated          SearchOrderBy = "updated_unix ASC"
    SearchOrderByRecentUpdated         SearchOrderBy = "updated_unix DESC, id DESC"
    SearchOrderByOldest                SearchOrderBy = "created_unix ASC"
    SearchOrderByNewest                SearchOrderBy = "created_unix DESC"
    SearchOrderBySize                  SearchOrderBy = "size ASC"
    SearchOrderBySizeReverse           SearchOrderBy = "size DESC"
    SearchOrderByID                    SearchOrderBy = "id ASC"
    SearchOrderByIDReverse             SearchOrderBy = "id DESC"
    SearchOrderByStars                 SearchOrderBy = "num_stars ASC"
    SearchOrderByStarsReverse          SearchOrderBy = "num_stars DESC"
    SearchOrderByForks                 SearchOrderBy = "num_forks ASC"
    SearchOrderByForksReverse          SearchOrderBy = "num_forks DESC"
)
```

But this may have unintended consequences.

Reply:
I agree the fixtures should have them, adding those now could potentially have unrelated effects/require unrelated changes I'd rather keep out of this already large PR.

Will look into it after this PR is merged though.